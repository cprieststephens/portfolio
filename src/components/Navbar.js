import React from "react";
import { Link } from "react-router-dom";
import "../styles/navbar.css";

const Navbar = () => {
  return (
    <div className="navbar">
      <ul className="navbar-links">
        <li>
          <Link className="navbar-links__item" to="/">
            About
          </Link>
        </li>
        <li>
          <Link className="navbar-links__item" to="projects">
            Projects
          </Link>
        </li>
      </ul>
    </div>
  );
};

export default Navbar;
