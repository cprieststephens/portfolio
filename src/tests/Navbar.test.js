import React from "react";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import Navbar from "../components/Navbar";

describe("Navbar", () => {
  it("renders correctly", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders about link text", () => {
    render(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>,
    );
    const about = screen.getByText(/about/i);

    expect(about).toBeInTheDocument();
  });

  it("renders projects link text", () => {
    render(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>,
    );
    const projects = screen.getByText(/projects/i);

    expect(projects).toBeInTheDocument();
  });
});
