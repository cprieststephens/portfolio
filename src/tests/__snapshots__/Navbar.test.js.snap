// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders correctly 1`] = `
<DocumentFragment>
  <div
    class="navbar"
  >
    <ul
      class="navbar-links"
    >
      <li>
        <a
          class="navbar-links__item"
          href="/"
        >
          About
        </a>
      </li>
      <li>
        <a
          class="navbar-links__item"
          href="/projects"
        >
          Projects
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
